modules/programs/go.nix

```diff
@@ -57,17 +57,15 @@ in
       home.file =
         let
           goPath = if cfg.goPath != null then cfg.goPath else "go";
-
-          mkSrc = n: v: {
-            target = "${goPath}/src/${n}";
-            source = v;
-          };
+          mkSrc = n: v: { "${goPath}/src/${n}".source = v; };
         in
-        mapAttrsToList mkSrc cfg.packages;
+          foldl' (a: b: a // b) {} (mapAttrsToList mkSrc cfg.packages);
```

Comment:
I wonder if the (a // b) may hide errors since duplicate overrides the previous one ?

Reply:
I believe it should not be an issue since the attribute sets being merged won't overlap due to the key in the attribute set produced by `mkSrc` including the key from the packages set.